lte/gateway/c/oai/tasks/mme_app/mme_app_bearer.c

```diff
@@ -1599,13 +1600,20 @@ void mme_app_handle_e_rab_setup_rsp(
     bearer_context_t* bc =
       mme_app_get_bearer_context(ue_context_p, (ebi_t) e_rab_id);
     if (bc->bearer_state & BEARER_STATE_SGW_CREATED) {
-      if (!(bc->bearer_state & BEARER_STATE_MME_CREATED)) {
+      if (bc->bearer_state & BEARER_STATE_MME_CREATED) {
```

Comment:
These statements (after the change) always are true (as the flags are set in mme_app_handle_s11_create_bearer_req).  We should remove them unless there is something we are missing.

Reply:
Above few lines are from legacy OAI codebase. These shall be relooked at the time of handling multiple bearers creation in single dedicated bearer create request.

Reply:
We can add comment saying that "Need to re-visit while handling multiple bearers creation in single dedicated bearer request"

Reply:
We can add comment saying that "Need to re-visit while handling multiple bearers creation in single dedicated bearer request"

Reply:
We can add comment saying that "Need to re-visit while handling multiple bearers creation in single dedicated bearer request"

Reply:
I don't understand the logic here. These statements will be always true, won't they? Is there a possibility that  BEARER_STATE_MME_CREATED and BEARER_STATE_SGW_CREATED are not set while we are in this function?

Reply:
How is multiple bearer will utilize this part?

Reply:
I took look on the multiple bearer handling in single create bearer request.  Looks like we can remove this parameter completely.
I shall make these changes and update the PR.

Reply:
> How is multiple bearer will utilize this part?
In case of multiple bearer handling in single create bearer request. from MME "Activate Dedicated EPS bearer context request" is sent to UE in "e_rab_setup_request" one after the other for all bearers, not in single message.
So MME shall wait for all responses sent to UE. Once all responses are received, MME shall send Dedicated Bearer Accept/Reject. Within this message MME shall include cause (success/reject cause) for each bearer. MME must have logic to identify what cause needs to be filled to each bearer. 
I was in the assumption that if bearer_state is ENB_CREATED then cause for that bearer would be successful and if it's only MME/SGW_CREATED, MME shall set the cause to "REQUEST_REJECTED"
But there is one parameter, bearer_status is there to judge, whether bearer is created successfully or not. Based on this information Dedicate  Bearer Accept/Reject is filled for each bearer.
So we don't need bearer_state. I can remove this parameter completely.
We can discuss in tomorrow's call as well.

Reply:
@rsarwad do you plan to make any changes to this PR?

Reply:
Yes Ulas. I made the changes and checked-in the changes

Reply:
Yes Ulas. I made the changes and checked-in the changes

Reply:
Yes Ulas. I made the changes and checked-in the changes